### DISCUSSION TITLE
[Feature] generic CSV import

### DISCUSSION THREAD

**Discussion by palitu** [johanohly/AirTrail#25] [Feature Request]
### I have searched the existing feature requests to make sure this is not a duplicate request.

- [X] Yes

### The feature

I would like to add a backlog of my flights into Airtrail, but i don't have them anywhere else.  It would be nice to be able to manually populate a generic CSV file, and upload it.

Maybe have an option to download a the headers then we can populate and import.  Maybe it would need some de-duplication and data checking, etc, but it looks like you have the majority of the code for importing, and it is just about setting up an AT csv and import process.

Thanks, this is a little bit of fun!

**Comment by johanohly** [comment]
Generic CSV importing is definitely planned, I like your idea of providing an "empty" CSV template for the sole purpose of adding flights quicker than through the UI.

**Comment by johanohly** [comment]
This has been implemented in #46. It will be available in the next release.

**Comment by Thomas1005** [comment]
At first, great work to all!
Is the CSV file import already working ?

**Reply by johanohly** [reply]
No, I haven't come around to it yet. You can however import AirTrail JSON files, whose format you can just kind of guess off of doing an export. 
I'll definitely come around to basic csv importing at some point along with a downloadable template.

**Reply by laneriddell** [reply]
Any updates to this?
I’m both a nerd and an airline pilot and think it would be cool to use a self-hosted mapping app like this to show my flights. I don’t personally track my flights anymore because my airline does that for me. I can download all my flights as a csv though, but obviously not in one of your formats. I figure I can just open it in Excel (or something), easily change it to a format with headers Airtrail likes, and resave it as a csv file.
Thanks for a cool program!
